src/jobservice/job/impl/gc/job.go

```diff
@@ -67,10 +70,16 @@ func (gc *GarbageCollector) Run(ctx env.JobContext, params map[string]interface{
 	if err := gc.init(ctx, params); err != nil {
 		return err
 	}
-	if err := gc.readonly(true); err != nil {
+	readOnlyCur, err := gc.getReadOnly()
+	if err != nil {
 		return err
 	}
-	defer gc.readonly(false)
+	if readOnlyCur != true {
+		if err := gc.setReadOnly(true); err != nil {
+			return err
+		}
+	}
+	defer gc.setReadOnly(readOnlyCur)
```

Comment:
Only needed when `readOnlyCur != true`, so should put it in the `if readOnlyCur != true` block.